src/jvmTest/kotlin/com/charleskorn/kaml/JvmYamlReadingTest.kt

```diff
@@ -42,5 +44,18 @@ class JvmYamlReadingTest : DescribeSpec({
                 result shouldBe 123
             }
         }
+
+        describe("reading an empty yaml without throwing an error") {
+            val input = ""
+            val bytes = ByteArrayInputStream(input.toByteArray(Charsets.UTF_8))
+
+            it("expect throwing an error because it's an empty document") {
+                shouldThrowExactly<EmptyYamlDocumentException> { Yaml.default.decodeFromStream<String>(bytes) }
+            }
+
+            it("expect ignoring empty document because of configuration") {
+                shouldNotThrowAny { Yaml(configuration = YamlConfiguration(allowReadingEmptyDocument = true)).decodeFromStream<String>(bytes) }
```

Comment:
In this scenario (reading a non-nullable string), what would you expect to be returned from `decodeFromStream`? Can we make an assertion that shows the expected behaviour?